clients/src/main/java/org/apache/kafka/clients/consumer/StickyAssignor.java

```diff
@@ -421,10 +515,17 @@ private int getBalanceScore(Map<String, List<TopicPartition>> assignment) {
             TreeSet<String> sortedConsumers = new TreeSet<>(new SubscriptionComparator(assignments));
             sortedConsumers.addAll(assignments.keySet());
 
+            List<TopicPartition> prevPartitions = new ArrayList(prevAssignment.keySet());
```

Comment:
Maybe I am wrong, but shouldn't this be moved inside the `while` loop, so it is initialized every time a new assignment is selected?
Because, now I think only the first consumer will be challenged against this previous partitions.

Reply:
You're right. The whole list needs to be compared against each consumer's assignment. Updated.